docs/azure-openshift.md

```diff
@@ -0,0 +1,165 @@
+```bash
+$ oc get secrets azure-credentials  -n kube-system  --template={{.data.azure_client_id}} | base64 -d
+$ oc get secrets azure-credentials  -n kube-system  --template={{.data.azure_client_secret}} | base64 -d
+$ oc get secrets azure-credentials  -n kube-system  --template={{.data.azure_region}} | base64 -d
+$ oc get secrets azure-credentials  -n kube-system  --template={{.data.azure_resource_prefix}} | base64 -d
+$ oc get secrets azure-credentials  -n kube-system  --template={{.data.azure_resourcegroup}} | base64 -d
+$ oc get secrets azure-credentials  -n kube-system  --template={{.data.azure_subscription_id}} | base64 -d
+$ oc get secrets azure-credentials  -n kube-system  --template={{.data.azure_tenant_id}} | base64 -d
+
+```
+
+4. Login to azure with base64 decoded values you get from above
+
+```bash
+$ az login --service-principal -u <azure_client_id> -p <azure_client_secret> --tenant <azure_tenant_id>
+```
+
+5. Get the routes to check the domain
+```bash
+oc get routes --all-namespaces | grep console
+openshift-console          console             console-openshift-console.apps.misalunk-azure.qe.azure.devcluster.openshift.com                       console             https   reencrypt/Redirect     None
+openshift-console          downloads           downloads-openshift-console.apps.misalunk-azure.qe.azure.devcluster.openshift.com                     downloads           http    edge/Redirect          None
+```
+
+6. Get the dns zone list w.r.t your azure_resourcegroup. Get it from the base64 decoded output.
+```bash
+$ az network dns zone list --resource-group <azure_resourcegroup>
+[]
+```
+Initially you will see nothing. So you will have create a zone for your resourcegroup.
+
+7. Create a dns zone for you resourcegroup.
+```bash
+ az network dns zone create -g <azure_resourcegroup> -n <pick up the zone name from the route  eg- the part after apps. eg - misalunk-azure.qe.azure.devcluster.openshift.com>
+```
+
+8. Then create CR as follows -
+```yaml
+apiVersion: externaldns.olm.openshift.io/v1alpha1
+kind: ExternalDNS
+metadata:
+  name: sample-azure
+spec:
+  domains:
+  - filterType: Include
+    matchType: Exact
+    name: test-azure1.qe.azure.devcluster.openshift.com
+  provider:
+    type: Azure
+  source:
+    # Incase you have multiple ingress controllers you must specify ingress controller name in the routerName
+    # so that the external dns will use the router canonical name correrponding to it to create a dns record.
+    openshiftRouteOptions:
+      routerName: default
+    #Source Type is route resource of OpenShift
+    fqdnTemplate:
+    - "{{.Name}}.apps.test-azure1.qe.azure.devcluster.openshift.com"
+    type: OpenShiftRoute
+  zones:
+  - "/subscriptions/53b4f551-f0fc-4bea-8cba-11111111111/resourceGroups/test-azure1-nxkxm-rg/providers/Microsoft.Network/dnszones/test-azure1.qe.azure.devcluster.openshift.com"
+```
+
+9. Create this CR using 
+```bash
+oc create -f <cr-file-name>.yaml
+```
```

Comment:
```suggestion
8. Then create CR as follows -
```yaml
$ cat <<EOF | oc create -f -
apiVersion: externaldns.olm.openshift.io/v1alpha1
kind: ExternalDNS
metadata:
  name: sample-azure
spec:
  domains:
  - filterType: Include
    matchType: Exact
    name: test-azure1.qe.azure.devcluster.openshift.com
  provider:
    type: Azure
  source:
    # Incase you have multiple ingress controllers you must specify ingress controller name in the routerName
    # so that the external dns will use the router canonical name correrponding to it to create a dns record.
    openshiftRouteOptions:
      routerName: default
    #Source Type is route resource of OpenShift
    fqdnTemplate:
    - "{{.Name}}.apps.test-azure1.qe.azure.devcluster.openshift.com"
    type: OpenShiftRoute
  zones:
  - "/subscriptions/53b4f551-f0fc-4bea-8cba-11111111111/resourceGroups/test-azure1-nxkxm-rg/providers/Microsoft.Network/dnszones/test-azure1.qe.azure.devcluster.openshift.com"
EOF
```
```